vividus-plugin-azure-cosmos-db/src/main/java/org/vividus/azure/cosmos/CosmosDbService.java

```diff
@@ -47,12 +48,21 @@ public CosmosContainer load(CosmosDbContainer cosmosDbContainer)
                         String accountKey = db.getAccountKey();
                         CosmosDbAccount account = accounts.get(accountKey,
                                 "Account configuration not found for the key: %s", accountKey);
-                        return new CosmosClientBuilder().endpoint(account.getEndpoint())
-                                                        .key(account.getKey())
-                                                        .connectionSharingAcrossClientsEnabled(true)
-                                                        .buildClient()
-                                                        .getDatabase(db.getId())
-                                                        .getContainer(cosmosDbContainer.getId());
+
+                        String key = account.getKey();
+                        CosmosClientBuilder builder = new CosmosClientBuilder().endpoint(account.getEndpoint())
+                                .connectionSharingAcrossClientsEnabled(true);
+                        if (key == null)
+                        {
+                            builder.credential(new DefaultAzureCredentialBuilder().build());
```

Comment:
it makes sense to inject already existing bean:
https://github.com/vividus-framework/vividus/blob/6cb56a0e07cc485c7604a116f75bc27722612eaf/vividus-extension-azure/src/main/resources/vividus-extension/spring.xml#L7
 instead of creation of new `TokenCredential`